docs/diagrams/wallet-lifecycle/wallet-lifecycle-1-handling-frauds.tikz

```diff
@@ -0,0 +1,187 @@
+                                                                                   \textbf{- LIVE\\
+                                                                                   - MOVING\_FUNDS\\
+                                                                                   - CLOSING}};
+\node[note2, above=of dfwt_note1.north, anchor=south, yshift=-8mm] (dfwt_note2)   {Sender incentivized by:\\ \textbf{Receiving recompensation from the treasury}};
+
+% Arrows
+% In places where arrows needs to bend more than once, we force the lines to
+% pass through intermediary points. 
+% (a,b) - sets point at an exact (a,b) coordinate
+% +(a,b) - sets point at current point and moves it by (a,b) vector, but doesn't
+%          remember the position afterwards
+% ++(a,b) - sets point at current point and moves it by (a,b) vector and
+%           remembers the position afterwards
+% To ease edition of the diagram in the future, usage of relative values is
+% recommended.
+\draw[->] (dfwt_start)   --                (dfwt_check1);
+\draw[->] (dfwt_check1)  -|                (dfwt_revert2) \no;
+\draw[->] (dfwt_check1)  -|                (dfwt_check2)  \yes;
+\draw[->] (dfwt_check2)  -|                (dfwt_check3)  \no;
+\draw[->] (dfwt_check2)  -|                (dfwt_revert1) \yes;
+\draw[->] (dfwt_check3)  -|                (dfwt_revert2) \no;
+\draw[->] (dfwt_check3)  -- +(4,0) \yes |- (df_action1);
+\draw[->] (df_action1)   --                (df_action2);
+
+
+% FRAUD DEFEAT TIMEOUT
+% Nodes
+\node[start, right=of dfwt_start, xshift=105mm]                   (fdt_start)   {Send notification about \textbf{fraud defeat timeout}};
+\node[check, below=of fdt_start]                                  (fdt_check1)  {Is there a pending\\ fraud notification that\\ hasn't been defeated?};
+\node[check, below right=of fdt_check1]                           (fdt_check2)  {Has the defeat \\ timeout passed?};
+\node[action, below right=of fdt_check2]                          (fdt_action1) {Return the challenger's deposit to him};
+\node[check, below=of fdt_action1]                                (fdt_check3)  {Is wallet LIVE,\\ MOVING\_FUNDS\\ or CLOSING?};
+\node[check, below left=of fdt_check3, xshift=-40mm]              (fdt_check4)  {Is wallet\\ TERMINATED?};
+\node[revert, below left=of fdt_check4]                           (fdt_revert1) {Transaction reverted};
+\node[action, below right=of fdt_check3]                          (fdt_action2) {Slashing, reward for the fraud challenge notifier};
+\node[check, below=of fdt_action2]                                (fdt_check5)  {Is wallet marked\\ as ACTIVE?};
+\node[action, below right=of fdt_check5]                          (fdt_action3) {Clear ACTIVE field};
+\node[action, below left=of fdt_action3, xshift=-20mm]            (fdt_action4) {Change wallet's state to TERMINATED, delete wallet from registry};
+\node[action, below=of fdt_action4]                               (fdt_action5) {Emit notification};
```

Comment:
Emitting an event is not that important, we emit a lot of events. Marking the challenge as resolved is very important though.
```suggestion
\node[action, below=of fdt_action4]                               (fdt_action5) {Resolve the challenge};
```